base/linalg/eigen.jl

```diff
@@ -90,11 +90,17 @@ end
 #Computes maximum and minimum eigenvalue
 function eigmax(A::Union{Number, StridedMatrix}; permute::Bool=true, scale::Bool=true)
     v = eigvals(A, permute = permute, scale = scale)
-    iseltype(v,Complex) ? error("DomainError: complex eigenvalues cannot be ordered") : maximum(v)
+    if iseltype(v,Complex)
+      throw(DomainError())
```

Comment:
The untested cholesky method sure (probably better to use `PosDefException` for that anyway), though the loss of information in this error (and the nonstandard indentation on this line) are unfortunate, and would be easy to forget to revisit. Maybe a TODO comment?